### DISCUSSION TITLE
grantScopes with an edgeObject

### DISCUSSION THREAD

**Discussion by TLadd** [hayes/pothos#774] [Q&A] [answered]
Is there a way to grant a scope to the node field of an edgeObject created using the Relay plugin? I attempted something like this:

``` 
const SomeEdge = builder.edgeObject({
  name: "SomeEdge",
  type: SomeType,
  grantScopes: () => ["readSomeType"],
});
```

but it doesn't seem to grant the scope to the node field.

**Comment by hayes** [comment, marked as answer]
interesting question. I don't think there is currently a way to do this.  The way to do this would be to add a `nodeField` options object to the edge options that supports adding options for the node field,  would be happy to accept a PR for this.  I can add this at some point, but might be a week or 2 before I have time

**Reply by hayes** [reply]
If you are interested in working on this, here are some relevant parts of the code:

the new option would be typed similar to https://github.com/hayes/pothos/blob/main/packages/plugin-relay/src/types.ts#L124-L136
and would be added to the options here: https://github.com/hayes/pothos/blob/main/packages/plugin-relay/src/global-types.ts#L373

and then the options would need to be spread in here: https://github.com/hayes/pothos/blob/main/packages/plugin-relay/src/schema-builder.ts#L554

**Reply by hayes** [reply]
the result would be something like:

```ts
const SomeEdge = builder.edgeObject({
  name: "SomeEdge",
  type: SomeType,
  nodeField: {
    grantScopes: () => ["readSomeType"],
  }
});
```

**Reply by hayes** [reply]
Ideally pothos would support this for `edges` on a connection as well

**Reply by TLadd** [reply]
Appreciate the pointers! Seems straightforward enough. I'll give it a go!
